zerver/views/auth.py

```diff
@@ -106,6 +115,39 @@ def create_preregistration_user(email: str, request: HttpRequest, realm_creation
         full_name_validated=full_name_validated,
     )
 
+def confirm_email_change(request: HttpRequest, confirmation_key: str,
```

Comment:
I moved this here because of some import errors caused due to importing `auth.py` functions in `user_settings.py` and vice versa.